internal/xds/translator/tracing.go

```diff
@@ -23,26 +24,61 @@ import (
 	"github.com/envoyproxy/gateway/internal/xds/types"
 )
 
+const (
+	envoyOpenTelemetry = "envoy.tracers.opentelemetry"
+	envoyZipkin        = "envoy.traces.zipkin"
+)
+
+type typConfigGen func() (*anypb.Any, error)
+
 func buildHCMTracing(tracing *ir.Tracing) (*hcm.HttpConnectionManager_Tracing, error) {
 	if tracing == nil {
 		return nil, nil
 	}
 
-	oc := &tracecfg.OpenTelemetryConfig{
-		GrpcService: &corev3.GrpcService{
-			TargetSpecifier: &corev3.GrpcService_EnvoyGrpc_{
-				EnvoyGrpc: &corev3.GrpcService_EnvoyGrpc{
-					ClusterName: tracing.Destination.Name,
-					Authority:   tracing.Authority,
+	var providerName string
+	var providerConfig typConfigGen
+
+	switch tracing.Provider.Type {
+	case egv1a1.TracingProviderTypeOpenTelemetry:
+		providerName = envoyOpenTelemetry
+
+		providerConfig = func() (*anypb.Any, error) {
+			config := &tracecfg.OpenTelemetryConfig{
+				GrpcService: &corev3.GrpcService{
+					TargetSpecifier: &corev3.GrpcService_EnvoyGrpc_{
+						EnvoyGrpc: &corev3.GrpcService_EnvoyGrpc{
+							ClusterName: tracing.Destination.Name,
+							Authority:   tracing.Authority,
+						},
+					},
 				},
-			},
-		},
-		ServiceName: tracing.ServiceName,
+				ServiceName: tracing.ServiceName,
+			}
+
+			return protocov.ToAnyWithError(config)
+		}
+	case egv1a1.TracingProviderTypeZipkin:
+		providerName = envoyZipkin
+
+		providerConfig = func() (*anypb.Any, error) {
+			config := &tracecfg.ZipkinConfig{
+				CollectorCluster:         tracing.Destination.Name,
+				CollectorEndpoint:        "/api/v2/spans",
+				TraceId_128Bit:           *tracing.Provider.Zipkin.Enable128BitTraceID,
```

Comment:
will crash if ptr is null

Reply:
you'll need to use something like `ptr.Deref(<>, false)` as well as on L69

Reply:
oversight on my part, should be handled now